frontend/pages/5_🐊_Investi_Gator.py

```diff
@@ -55,6 +56,41 @@ def reset_state():
     """
     )
 
+    if not st.session_state.get("token"):
```

Comment:
Any way we can centralize this streamlit part in a separate file and import it instead of dupliating? Not sure how that would work in streamlit, not worth spending a lot of time on I think, just a thought.

Reply:
Yeah - repeating the authentication logic throughout the app is definitely very hacky, I was running into issues where navigating from one page to the other would sometimes refresh the state and lose the token. Can definitely look into centralizing the code!